compiler/semexprs.nim

```diff
@@ -627,7 +627,9 @@ proc analyseIfAddressTakenInCall(c: PContext, n: PNode) =
   # it may be a proc var with a generic alias type, so we skip over them
   var t = n.sons[0].typ.skipTypes({tyGenericInst, tyAlias, tySink})
 
-  if n.sons[0].kind == nkSym and n.sons[0].sym.magic in FakeVarParams:
+  if n.sons[0].kind == nkSym and (
+    n[0].sym.magic in FakeVarParams or
+    sfBorrow in n[0].sym.flags and n[0].sym.getBody.sym.magic in FakeVarParams):
```

Comment:
Shouldn't instead ``.magic`` borrowed procs keep the magic field?

Reply:
That works too, it's not completely correct IMO since the magic is attached to the symbol you borrow from and not the borrowed one but I'm nitpicking here.